codecov.yaml

```diff
@@ -0,0 +1,22 @@
+coverage:
+  status:
+    project:
+      default:
+        target: auto
+    patch:
+      default:
+        target: auto
+
+comment:
+  show_carryforward_flags: true
+
+flag_management:
+  default_rules: # the rules that will be followed for any flag added, generally
+    carryforward: true
+    statuses:
+      - name_prefix: project-
+        type: project
+        target: auto
+      - name_prefix: patch-
+        type: patch
+        target: auto
```

Comment:
Could you explain all other parameters for other people?

Reply:
## `coverage` section
`coverage` section defines project and patch statuses.

- `project` means the whole coverage change on the repository.
- `patch` means the differential coverage change on the pull request or single commit.

`default` defines default target value. If you need to modify it for specific flags, you can add new keys.

Each section has `target` key.
Target can choose a minimum coverage ratio that the commit must meet to be considered a success.
`auto` will use the coverage from the base commit (pull request base or parent commit) coverage to compare against.

Reference:
https://docs.codecov.com/docs/commit-status#project-status

## `comment` section

By default, any flags that are carried forward will not show up in the pull request / merge request in your code host (Github, Gitlab, or Bitbucket) in the status checks and/or the pull request comments.

`show_carryforward_flags: true`: if any flag coverage was carried forward, the flags table will have an additional column labeled “Carriedforward” indicating which flags had coverage carried forward

Reference:
https://docs.codecov.com/docs/carryforward-flags#carryforward-flags-in-the-pull-request-comment

## `flag_management` section

This is a recommended approach to manage flags.
https://docs.codecov.com/docs/flags#recommended-automatic-flag-management

`default_rules` defines default rules for all flags.
`carryforward`: All flags are assumed to carry forward flags.

`statuses`: Report coverage statuses for each flag. The target and type are the same as `coverage` section. `name_prefix` adds prefix for each flag.
![image](https://user-images.githubusercontent.com/19993104/160998096-dffbd673-72ee-450d-b0b2-92ec89ea62c1.png)